sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/ResourceExtensionsTests.cs

```diff
@@ -8,236 +8,312 @@
-            Assert.DoesNotContain("_", SdkVersionUtils.s_sdkVersion);
+        var monitorBase = azMonResource.MetricTelemetry.Data;
+        var metricsData = monitorBase.BaseData as MetricsData;
 
-            // Clean up
-            SdkVersionUtils.s_sdkVersion = sdkVersion;
-        }
+        var metricDataPoint = metricsData?.Metrics[0];
+        Assert.Equal("bar", metricsData?.Properties["foo"]);
+        Assert.False(metricsData?.Properties.ContainsKey("ai.sdk.prefix"));
 
-        [Fact]
-        public void SdkPrefixIsNotInResourceMetrics()
+        // Clean up
+        SdkVersionUtils.s_sdkVersion = sdkVersion;
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData(InstrumentationKey)]
+    public void MetricTelemetryHasAllResourceAttributes(string? instrumentationKey)
+    {
+        var testAttributes = new Dictionary<string, object>
         {
-            // SDK version is static, preserve to clean up later.
-            var sdkVersion = SdkVersionUtils.s_sdkVersion;
-            var testAttributes = new Dictionary<string, object>
-            {
-                {"foo", "bar" },
-                { "ai.sdk.prefix", "pre_" }
-            };
+            {SemanticConventions.AttributeServiceName, "my-service" },
+            {SemanticConventions.AttributeServiceNamespace, "my-namespace" },
+            {SemanticConventions.AttributeServiceInstance, "my-instance" },
+            { "foo", "bar" }
+        };
 
-            var resource = ResourceBuilder.CreateDefault().AddAttributes(testAttributes).Build();
-            var azMonResource = resource.CreateAzureMonitorResource(InstrumentationKey);
+        var resource = ResourceBuilder.CreateEmpty().AddAttributes(testAttributes).Build();
+        var azMonResource = resource.CreateAzureMonitorResource(instrumentationKey);
 
+        Assert.Equal(instrumentationKey != null, azMonResource?.MetricTelemetry != null);
+
+        if (instrumentationKey != null)
+        {
             Assert.Equal("Metric", azMonResource!.MetricTelemetry!.Name);
+            Assert.Equal(3, azMonResource.MetricTelemetry.Tags.Count);
+            Assert.NotNull(azMonResource.MetricTelemetry.Data);
 
             var monitorBase = azMonResource.MetricTelemetry.Data;
             var metricsData = monitorBase.BaseData as MetricsData;
 
+            Assert.NotNull(metricsData?.Metrics);
+
             var metricDataPoint = metricsData?.Metrics[0];
-            Assert.Equal("bar", metricsData?.Properties["foo"]);
-            Assert.False(metricsData?.Properties.ContainsKey("ai.sdk.prefix"));
+            Assert.Equal("_OTELRESOURCE_", metricDataPoint?.Name);
+            Assert.Equal(0, metricDataPoint?.Value);
 
-            // Clean up
-            SdkVersionUtils.s_sdkVersion = sdkVersion;
+            Assert.Equal(4, metricsData?.Properties.Count);
+
+            Assert.Equal("my-service", metricsData?.Properties[SemanticConventions.AttributeServiceName]);
```

Comment:
Could we add the kubernetes attributes here so to test we send them as custom metric even if they are not used in cloud.role and roleinstance?